source/core/grids.cc

```diff
@@ -417,26 +417,100 @@ read_mesh_and_manifolds_for_stator_and_rotor(
   GridTools::rotate(mortar_parameters.rotor_mesh->rotation_angle,
                     rotor_temp_tria);
 
-  // Check manifold IDs at rotor boundary
-  std::set<unsigned int> rotor_boundary_manifold_ids;
+  // Get stator manifold ids without flat id
+  unsigned int stator_ids_no_flat = 0;
+  for (const auto &id : stator_temp_tria.get_manifold_ids())
+    {
+      if (id != numbers::flat_manifold_id)
+        stator_ids_no_flat++;
+    }
+
```

Comment:
I think this would be the place to check `stator_ids_no_flat == len(stator_temp_tria.get_manifold_ids() without flat_manifold_id)`. Similar a check for the boundary ids would be good. Better be safe!

Reply:
Agreed, this is the kind of thing that can prevent a lot of problems down the road